tests/func/test_api.py

```diff
@@ -126,3 +128,39 @@ def test_open_not_cached(dvc):
     os.remove(metric_file)
     with pytest.raises(FileMissingError):
         api.read(metric_file)
+
+
+def test_summon(tmp_dir, erepo_dir, dvc, monkeypatch):
+    artifacts_yaml = ruamel.yaml.dump(
+        {
+            "artifacts": [
+                {
+                    "name": "sum",
+                    "description": "Add <x> to <number>",
+                    "file": "calculator.py",
+                    "method": "add_to_num",
+                    "params": {"x": 1},
+                    "deps": ["number"],
+                }
+            ]
+        }
+    )
+
+    # XXX: Chdir because `stage.create` defaults to `wdir = os.curdir`
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+
+        erepo_dir.dvc_gen("number", "100")
+
+    erepo_dir.scm_gen("artifacts.yaml", artifacts_yaml)
+    erepo_dir.scm_gen(
+        "calculator.py",
+        "def add_to_num(x): return x + int(open('number').read())",
+    )
+    erepo_dir.scm.add(["number.dvc", "calculator.py", "artifacts.yaml"])
+    erepo_dir.scm.commit("Add files")
```

Comment:
This probably won't work without a chdir also. Also, `.scm_gen()` already does `.scm.add()`.

Reply:
should we do something to address the `chdir` requirement in those cases or is it ok to just monkeypatch it?

Reply:
See https://github.com/iterative/dvc/issues/2967, we chdir for now. Will probably do that anyway to have less magic.